crypto/blake2/blake2s.c

```diff
@@ -79,6 +87,11 @@ int BLAKE2s_Init(BLAKE2S_CTX *c)
 {
```

Comment:
Same issues as in `blake2b` 

I hate this comment.

We're now returning a zero but it seems that the return is expected a qualified structure that may be just randomly allocated space and allow the other functions like `blake2b_compress` to run unexpectedly with random input.

See the path from `blake2b_init_param` to `blake2_init0` that is done prior to the `return 1`

We might need to add the fips check to `static void black2b_compress`
```
    if (FIPS_mode()) {
        FIPSerr(ERR_LIB_FIPS, FIPS_R_NON_FIPS_METHOD);
        return 0;
    }
```

Reply:
Again. Static functions. "You can't get there from here". All entry points are covered with OpenSSLdie or error returns already.

Reply:
Yeah its called at the end of this `init` function.

I missed the static part of the compress when I saw that the fips check was missing and the weird statement in the comment that the `inti` will always return 1 then saw that `BLAKE2B_CTX *c` was memset with in `int BLAKE2b_Init(BLAKE2B_CTX *c)` that I then forgot the start of the fuction and jumped into how it could have been used in that function.

resolving